express-notes-api/index.js

```diff
@@ -0,0 +1,93 @@
+const express = require('express');
+const data = require('./data.json');
+const app = express();
+
+const fs = require('fs');
+
+const notes = data.notes;
+let nextId = data.nextId;
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res, next) => {
+  const notesArray = [];
+  for (const key in notes) {
+    const noteObjects = notes[key];
+    notesArray.push(noteObjects);
+  }
+  res.status(200).json(notesArray);
+  next();
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const num = parseInt(req.params.id, 10);
+  if (!Number.isInteger(num)) {
+    res.status(400).json({ error: 'id must be a positive integer' });
```

Comment:
Any non-positive-integer id should result in a `400 Bad Request` (for example, `GET /api/notes/-4353.354ugfdgjk`).